cmd/gitops-server/cmd/cmd.go

```diff
@@ -78,28 +84,29 @@ func NewCommand() *cobra.Command {
 
 	options = Options{}
 
-	cmd.Flags().StringVar(&options.LogLevel, "log-level", logger.DefaultLogLevel, "log level")
+	// System config
 	cmd.Flags().StringVar(&options.Host, "host", server.DefaultHost, "UI host")
-	cmd.Flags().StringVar(&options.Port, "port", server.DefaultPort, "UI port")
-	cmd.Flags().StringVar(&options.Path, "path", "", "Path url")
+	cmd.Flags().StringVar(&options.LogLevel, "log-level", logger.DefaultLogLevel, "log level")
 	cmd.Flags().StringVar(&options.NotificationControllerAddress, "notification-controller-address", "", "the address of the notification-controller running in the cluster")
-
-	cmd.Flags().StringVar(&options.TLSCertFile, "tls-cert-file", "", "filename for the TLS certificate, in-memory generated if omitted")
-	cmd.Flags().StringVar(&options.TLSKeyFile, "tls-private-key-file", "", "filename for the TLS key, in-memory generated if omitted")
+	cmd.Flags().StringVar(&options.Path, "path", "", "Path url")
+	cmd.Flags().StringVar(&options.Port, "port", server.DefaultPort, "UI port")
+	cmd.Flags().StringSliceVar(&options.AuthMethods, "auth-methods", auth.DefaultAuthMethodStrings(), "Which auth methods to use, valid values are 'oidc' and 'user-account'")
```

Comment:
I kinda feel this should be set through a set of env vars rather than through one command line arg. Main reason being, it composes easily, and doesn't require any changes to the CLI interface.

Reply:
I think in general I agree but I think that should be part of a piece of work to move most of the configuration to envars rather than our current CLI heavy system. I can add it for these if you want but it does feel a little out of scope?

Reply:
I didn't mean to add env vars to these command line args - I meant to remove the command line args. `NewAuthServerConfig` could call a function that does `os.GetEnv` three times, and thus avoid having to change the API as much.